pkg/kn/commands/plugin/path_verifier_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright © 2018 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/knative/client/pkg/kn/commands"
+	"github.com/spf13/cobra"
+	"gotest.tools/assert"
+)
+
+var (
+	pluginPath string
+	rootCmd    *cobra.Command
+	verifier   *CommandOverrideVerifier
+)
+
+var setup = func(t *testing.T) {
+	knParams := &commands.KnParams{}
+	rootCmd, _, _ = commands.CreateTestKnCommand(NewPluginCommand(knParams), knParams)
+	verifier = &CommandOverrideVerifier{
+		Root:        rootCmd,
+		SeenPlugins: make(map[string]string),
+	}
+}
+
+var cleanup = func(t *testing.T) {
+	if pluginPath != "" {
+		DeleteTestPlugin(t, pluginPath)
+	}
+}
+
+func TestCommandOverrideVerifier_Verify_with_nil_root_command(t *testing.T) {
+	t.Run("returns error verifying path", func(t *testing.T) {
```

Comment:
Technically this level of indirection isn't needed - you can just do everything in this Func at the top level and get basically the same results

Reply:
I want to separate the different context w/o having to write multiple tests